components/brave_wallet/browser/zcash/rust/lib.rs

```diff
@@ -56,13 +74,78 @@ macro_rules! impl_result {
     };
 }
 
+#[derive(Clone)]
+pub(crate) struct MockRng(u64);
+
+impl CryptoRng for MockRng {}
+
+impl RngCore for MockRng {
+    fn next_u32(&mut self) -> u32 {
+        self.next_u64() as u32
+    }
+
+    fn next_u64(&mut self) -> u64 {
+        self.0 += 1;
+        self.0
+    }
+
+    fn fill_bytes(&mut self, dest: &mut [u8]) {
+        // https://github.com/rust-random/rand/blob/master/rand_core/src/impls.rs#L38
+        let mut left = dest;
+        while left.len() >= 8 {
+            let (l, r) = { left }.split_at_mut(8);
+            left = r;
+            let chunk: [u8; 8] = self.next_u64().to_le_bytes();
+            l.copy_from_slice(&chunk);
+        }
+        let n = left.len();
+        if n > 4 {
+            let chunk: [u8; 8] = self.next_u64().to_le_bytes();
+            left.copy_from_slice(&chunk[..n]);
+        } else if n > 0 {
+            let chunk: [u8; 4] = self.next_u32().to_le_bytes();
+            left.copy_from_slice(&chunk[..n]);
+        }
+    }
+
+    fn try_fill_bytes(&mut self, dest: &mut [u8]) -> Result<(), OtherError> {
+        Ok(self.fill_bytes(dest))
+    }
+}
+
+
 #[allow(unsafe_op_in_unsafe_fn)]
 #[cxx::bridge(namespace = brave_wallet::orchard)]
```

Comment:
it's brave_wallet::orchard::OrchardOutput, but there's also brave_wallet::OrchardOutput and ZCashTransaction::OrchardOutput?

Reply:
brave_wallet::orchard::OrchardOutput is a ffi structure to pass outputs to rust crate.
Removed ZCashTransaction::OrchardOutput since it is identical to  brave_wallet::OrchardOutput